docs/src/framework/a-100-line-kernel.md

```diff
@@ -7,23 +7,28 @@ we will show a new kernel in about 100 lines of safe Rust.
 Our new kernel will be able to run the following Hello World program.
 
 ```s
-global _start
-
-section .text
-
+.global _start                      # entry point
+.section .text                      # code section
 _start:
-    mov rax, 1        ; syswrite
-    mov rdi, 1        ; fd
-    mov rsi, msg      ; "Hello, world!\n",
-    mov rdx, msglen   ; sizeof("Hello, world!\n")
-    syscall
-
-    mov rax, 60       ; sys_exit
-    mov rdi, 0        ; exit_code
+    mov     $1, %rax                # syscall number of write
+    mov     $1, %rdi                # stdout
+    mov     $message, %rsi          # address of message         
+    mov     $message_end, %rdx
+    sub     %rsi, %rdx              # calculate message len
     syscall
+    mov     $60, %rax               # syscall number of exit, move it to rax
+    mov     $0, %rdi                # exit code, move it to rdi
+    syscall  
+
+.section .rodata                    # read only data section
+message:
+    .ascii  "Hello, world\n"
+message_end:
+```
 
-section .rodata
-    msg: db "Hello, world!", 10
+Use the following command to compile the program:
```

Comment:
To make the flow of the text more smooth, I rewrite the original sentence by beginning the sentence with the _program_ (the subject before the sentence) and ending it with the _command_ (the subject after the sentence).

> The assembly program above can be compiled with the following command.